crates/storage/provider/src/providers/state/latest.rs

```diff
@@ -118,8 +127,17 @@ impl<Provider: DBProvider + StateCommitmentProvider> StorageRootProvider
         slots: &[B256],
         hashed_storage: HashedStorage,
     ) -> ProviderResult<StorageMultiProof> {
-        StorageProof::overlay_storage_multiproof(self.tx(), address, slots, hashed_storage)
-            .map_err(ProviderError::from)
+        StorageProof::overlay_storage_multiproof(
+            &StorageProof::new(
+                DatabaseTrieCursorFactory::new(self.tx()),
+                DatabaseHashedCursorFactory::new(self.tx()),
+                address,
+            ),
+            address,
+            slots,
+            hashed_storage,
```

Comment:
imo we should implement a method on `StorageProof` that is identical to the previous `from_tx` implementation, then we should be able to do something like:
```
StorageProof::from_tx(self.tx()).overlay_storage_multiproof(address, slots, hashed_storage)
```